### ISSUE TITLE
Investigate terminating connections in sdap_ops.c and comment the code some more

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1511]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1507
- Created at 2012-09-05 12:05:20 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed at 2020-03-24 14:12:29 as wontfix
- Assigned to nobody
---

We removed a piece of code that was causing a double-free in 52828e4cc7f980f9e41fb604270a90b5e197c806. 

During the review, the following was recommended by the original developer:

    The main point of this code was to prevent deleting the connection every time  
    operation ends. The patch introducing the disconnection flag should be solving 
    two major things:
    
    Don't accept new operations for the connection that is disconnecting. This is  
    solved in another part of the code.
    
    Don't free the connection immediatelly when you detect it's not working. 
    Rather mark it as disconnecting and let all operations on that connection 
    finish (probably with an error). Only the last finished operation should be 
    deleting the connection. Otherwise you will sometimes get hanging operations.  
    IIRC, that's what I wanted to solve by this hunk of the code. I probably 
    didn't notice the code deleting connection every time the operation ends.

We should investigate the above and possibly fix the disconnection logic.

At the same time, it is evident that this whole module is hard to
understand. We should make sure it is more readable by introducing more
comments or (careful) refactoring.

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-09-06 15:30:00](https://pagure.io/SSSD/sssd/issue/1507#242362)**

We need to investigate this code more. A separate ticket should be filed to add more comments to the code.

description: We removed a piece of code that was causing a double-free in 52828e4cc7f980f9e41fb604270a90b5e197c806. 

During the review, the following was recommended by the original developer:
{{{
The main point of this code was to prevent deleting the connection every time  
operation ends. The patch introducing the disconnection flag should be solving 
two major things:

Don't accept new operations for the connection that is dicsonnecting. This is  
solved in another part of the code.

Don't free the connection immediatelly when you detect it's not working. 
Rather mark it as disconnecting and let all operations on that connection 
finish (probably with an error). Only the last finished operation should be 
deleting the connection. Otherwise you will sometimes get hanging operations.  
IIRC, that's what I wanted to solve by this hunk of the code. I probably 
didn't notice the code deleting connection every time the operation ends.
}}}

We should investigate the above and possibly fix the disconnection logic.

At the same time, it is evident that this whole module is hard to
understand. We should make sure it is more readable by introducing more
comments or (careful) refactoring.
 => We removed a piece of code that was causing a double-free in 52828e4cc7f980f9e41fb604270a90b5e197c806. 

During the review, the following was recommended by the original developer:
{{{
The main point of this code was to prevent deleting the connection every time  
operation ends. The patch introducing the disconnection flag should be solving 
two major things:

Don't accept new operations for the connection that is disconnecting. This is  
solved in another part of the code.

Don't free the connection immediatelly when you detect it's not working. 
Rather mark it as disconnecting and let all operations on that connection 
finish (probably with an error). Only the last finished operation should be 
deleting the connection. Otherwise you will sometimes get hanging operations.  
IIRC, that's what I wanted to solve by this hunk of the code. I probably 
didn't notice the code deleting connection every time the operation ends.
}}}

We should investigate the above and possibly fix the disconnection logic.

At the same time, it is evident that this whole module is hard to
understand. We should make sure it is more readable by introducing more
comments or (careful) refactoring.

milestone: NEEDS_TRIAGE => SSSD 1.9.1
rhbz:  => 0
type: defect => task

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-10-05 17:31:42](https://pagure.io/SSSD/sssd/issue/1507#242363)**

Fields changed

milestone: SSSD 1.9.1 => SSSD 1.9.2

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-10-09 11:12:38](https://pagure.io/SSSD/sssd/issue/1507#242364)**

Fields changed

owner: somebody => okos
status: new => assigned

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-10-11 15:20:12](https://pagure.io/SSSD/sssd/issue/1507#242365)**

Fields changed

milestone: SSSD 1.9.2 => SSSD 1.9.3

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-11-08 12:59:48](https://pagure.io/SSSD/sssd/issue/1507#242366)**

Not critical for 1.9.3

design:  => 
design_review:  => 0
fedora_test_page:  => 
milestone: SSSD 1.9.3 => SSSD 1.9.4

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-11-29 10:57:05](https://pagure.io/SSSD/sssd/issue/1507#242367)**

Dropping the investigation/documentation tasks to trivial. These can be deferred if needed.

priority: major => trivial

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-12-16 14:18:15](https://pagure.io/SSSD/sssd/issue/1507#242368)**

Moving docs task to 1.9.5

milestone: SSSD 1.9.4 => SSSD 1.9.5

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-03-21 18:47:30](https://pagure.io/SSSD/sssd/issue/1507#242369)**

Fields changed

milestone: SSSD 1.9.5 => SSSD 1.11 beta
review:  => 0
selected:  =>

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-07-30 12:34:21](https://pagure.io/SSSD/sssd/issue/1507#242370)**

Fields changed

changelog:  => 
milestone: SSSD 1.13 beta => Interim Bucket
priority: trivial => major

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-07-30 12:54:43](https://pagure.io/SSSD/sssd/issue/1507#242371)**

Fields changed

milestone: Interim Bucket => SSSD 1.12 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-05-30 16:05:18](https://pagure.io/SSSD/sssd/issue/1507#242372)**

Should be done with the LDAP refactoring planned for 1.13 

milestone: SSSD 1.12 beta => SSSD 1.13 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-09-26 18:49:56](https://pagure.io/SSSD/sssd/issue/1507#242373)**

Fields changed

mark:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-10 16:32:36](https://pagure.io/SSSD/sssd/issue/1507#242374)**

This is a prerequisite of refactoring the LDAP provider. We need to have the whole codebase documented first.

milestone: SSSD 1.13 beta => SSSD 1.13 backlog

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-12 20:12:50](https://pagure.io/SSSD/sssd/issue/1507#242375)**

Mass-moving tickets not planned for the 1.13 release to 1.14

milestone: SSSD 1.13 backlog => SSSD 1.14 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-06-20 11:48:57](https://pagure.io/SSSD/sssd/issue/1507#242376)**

The sdap_ops.c module still needs a lot of work..

milestone: SSSD 1.14 beta => SSSD 1.15 beta
sensitive:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-10-05 22:35:29](https://pagure.io/SSSD/sssd/issue/1507#242377)**

Fields changed

milestone: SSSD 1.16 beta => SSSD 1.15 Alpha
owner: okos => somebody
status: assigned => new

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-01-25 12:16:45](https://pagure.io/SSSD/sssd/issue/1507#242378)**

Fields changed

milestone: SSSD 1.15.0 => SSSD 1.15 Beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-06 15:40:54](https://pagure.io/SSSD/sssd/issue/1507#242379)**

Fields changed

milestone: SSSD 1.15.3 => SSSD 1.15.2

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-24 15:06:40](https://pagure.io/SSSD/sssd/issue/1507#242380)**

**Metadata Update from @jhrozek**:
- Issue marked as blocked by: #2976
- Issue set to the milestone: SSSD 1.15.2

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-03 14:16:12](https://pagure.io/SSSD/sssd/issue/1507#429287)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch reset
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-03-15 10:36:40](https://pagure.io/SSSD/sssd/issue/1507#431423)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch reset
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset
- Issue set to the milestone: SSSD Future releases (no date set yet) (was: SSSD 1.15.2)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-13 14:23:16](https://pagure.io/SSSD/sssd/issue/1507#633190)**

**Metadata Update from @thalman**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue tagged with: Canditate to close

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:12:28](https://pagure.io/SSSD/sssd/issue/1507#636698)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:12:30](https://pagure.io/SSSD/sssd/issue/1507#636699)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
